### PR TITLE
Remove Reanimated 1 from ButtonPressAnimation and migrate to TS

### PR DESCRIPTION
**Pull request by jkadamczyk**
Fixes RNBW-3382

## What changed (plus any additional context for devs)
- Rewrote the component to TS
- Consulted Reanimated 1 usage on iOS with Michał – Reanimated 1 implementation of a button was left due to support for legacy versions of the app (there is also a Native button implementation), it was long time ago so we should be able to safely remove the old implementation.
- Bumped Reanimated to 2.8.0 which is the last version before v3, it resolved some of our typings issues

### iOS:
before:
https://streamable.com/zrc3c3
after:
https://streamable.com/y7f5z5

### Android:
before:
https://streamable.com/bdip4h
after:
https://streamable.com/9ecf6e

## Dev checklist for QA: what to test
* Generally check if buttons are not behaving in a weird way, the changes I made have impact on almost everything that is touchable with the zoom effect

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [x] Added e2e tests? if not please specify why – No functionality was changed
- [x] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: src/components/animations/ButtonPressAnimation/ScaleButtonZoomable.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}: Props) => {
+  const scale = useSharedValue(1);
+  const scaleTraversed = useDerivedValue(() => {
+    const value = withTiming(scale.value, {
+      duration,
+      easing: Easing.bezier(0.25, 0.1, 0.25, 1),
+    });
+    return value;
+  });
+  const sz = useAnimatedStyle(() => {
```
Comment (tchayen) [inline src/components/animations/ButtonPressAnimation/ScaleButtonZoomable.tsx:39]: This name is a bit confusing to me. What do you think about `scaleZ` (if I understood correctly what it means)?

Comment (jkadamczyk) [reply src/components/animations/ButtonPressAnimation/ScaleButtonZoomable.tsx:39]: I will rename it, TBH I just copied the JS code and filled in the interfaces 😄

**Comment by jkadamczyk** [comment]
This one will need a full regression because:
A) bumped Reanimated version
B) Touched a heavily used component that sits in a lot of places

File: src/design-system/docs/README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ yarn dev
 
 Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
 
-You can start editing the page by modifying `pages/index.tsx`. The page auto-updates as you edit the file.
+You can start editing the page by modifying `pages/index.ts`. The page auto-updates as you edit the file.
```
Comment (tchayen) [inline src/design-system/docs/README.md:15]: Are you sure it's supposed to be here?

File: patches/react-native-reanimated+2.8.0.patch [hunk 3]
```diff
@@ -0,0 +1,1291 @@
...
++    reattachNeeded: boolean;
++    listeners: Record<string, (event: T) => void>;
++    viewTag: number | undefined;
++    registrations: string[];
++    constructor(worklet: (event: T) => void, eventNames?: string[]);
++    updateWorklet(newWorklet: (event: T) => void): void;
++    registerForEvents(viewTag: number, fallbackEventName?: string): void;
++    unregisterFromEvents(): void;
++  }
+ }
```
Comment (brunobar79) [inline patches/react-native-reanimated+2.8.0.patch:1291]: The patch is quite big (1.2k lines). Can you explain why are all this changes needed?

**Review by brunobar79 — COMMENTED** [review]
Looks good to me!  I'd like to see CI passing before approving

File: patches/react-native-reanimated+2.8.0.patch (hunk 3 above)
Comment (jkadamczyk) [reply patches/react-native-reanimated+2.8.0.patch:1291]: I can't need to take a look at it, the only thing that should be changed is adding one class from previous patch. I will fix it, thanks for spotting that!

Comment (jkadamczyk) [reply patches/react-native-reanimated+2.8.0.patch:1291]: I think I ran code formatter 🤦

File: src/utils/haptics.ts [hunk 4]
```diff
@@ -11,14 +11,14 @@ export const HapticFeedbackTypes = {
...
+export type HapticFeedbackType = typeof HapticFeedback[keyof typeof HapticFeedback];
+
+const hapticToTrigger = (haptic: HapticFeedbackType) => ({
   [haptic]: () => ReactNativeHapticFeedback.trigger(haptic),
 });
 
-const haptics = reduceArrayToObject(
-  map(keys(HapticFeedbackTypes), hapticToTrigger)
-);
+const haptics = reduceArrayToObject(map(keys(HapticFeedback), hapticToTrigger));
```
Comment (osdnk) [inline src/utils/haptics.ts:22]: can you use regular JS features here?

Comment (jkadamczyk) [reply src/utils/haptics.ts:22]: Yep, I think so

**Comment by jkadamczyk** [comment]
I think I managed to fix e2e, it was duplicated `testID` in `Button.ios.js` I used spread operator with it on both `ButtonPressAnimation` and `Container` (View) which made detox hang at some point

**Comment by jkadamczyk** [comment]
CI still fails but somewhere else, will take a look tomorrow

**Review by osdnk — APPROVED** [review]

**Review by skylarbarrera — APPROVED** [review]

**Review by estebanmino — APPROVED** [review]

**Review by ibrahimtaveras00 — APPROVED** [review]

**Pull request merged**